.github/workflows/macos.yml

```diff
@@ -35,34 +35,35 @@ env:
   Z3_VERSION: 4.13.0
 
 jobs:
-  build_and_test:
+  build_and_test_other_macos:
     strategy:
       matrix:
-        os: [macos-13, macos-14]
-        compiler: [g++-12, g++-13, g++-14, clang++]
+        os: [macos-13, macos-14, macos-15]
+        architecture: [x64, arm64]
+        compiler: [g++-13, g++-14, clang++]
         include:
           - os: macos-13
             architecture: x64
           - os: macos-14
             architecture: arm64
+          - os: macos-15
+            architecture: arm64
           - compiler: clang++
             ccompiler: clang
-          - compiler: g++-12
-            ccompiler: gcc-12
           - compiler: g++-13
             ccompiler: gcc-13
           - compiler: g++-14
             ccompiler: gcc-14
         exclude:
           - os: macos-14
-            compiler: g++-12
-          - os: macos-14
-            compiler: g++-13
-          - os: macos-14
-            compiler: g++-14
+            architecture: x64
+          - os: macos-15
+            architecture: x64
+          - os: macos-13
+            architecture: arm64
 
-    name: 🍎 ${{matrix.os}} with ${{matrix.compiler}}
-    runs-on: ${{matrix.os}}
+    name: 🍎 ${{ matrix.os }} with ${{ matrix.compiler }} with ${{ matrix.architecture }}
```

Comment:
The added spaces here pose an inconsistency, or? I thought `pre-commit` would automatically address this 🤔

Reply:
Yes, weird! I also assumed that ``pre-commit`` would fix it.